product_dimension/models/product.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# © 2015 ADHOC SA  (http://www.adhoc.com.ar)
+# © 2015-2016 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from openerp import models, fields
+from openerp import api
+
+
+class Product(models.Model):
+    _inherit = 'product.product'
+
+    @api.onchange('length', 'height', 'width', 'dimensional_uom_id')
+    def onchange_calculate_volume(self):
+        if (not self.length or not self.height or not self.width or
+                not self.dimensional_uom_id):
+            return False
+
+        length_m = self.convert_to_meters(self.length, self.dimensional_uom_id)
+        height_m = self.convert_to_meters(self.height, self.dimensional_uom_id)
+        width_m = self.convert_to_meters(self.width, self.dimensional_uom_id)
+        self.volume = length_m * height_m * width_m
+
+    def convert_to_meters(self, measure, dimensional_uom):
+        uom_meters = self.env['product.uom'].search([('name', '=', 'm')])
+
+        return self.env['product.uom']._compute_qty_obj(
+            from_unit=dimensional_uom,
+            qty=measure,
+            to_unit=uom_meters)
+
+    @api.model
+    def _get_dimension_uom_domain(self):
+        return [
+            ('category_id', '=', self.env.ref('product.uom_categ_length').id)
+        ]
+
+    length = fields.Float()
+    height = fields.Float()
+    width = fields.Float()
```

Comment:
I'm wondering if we want to have a decimal precision here (this is really a question).